### PR TITLE
Implement Bindless Textures on Shader Decompiler and GL backend

### PR DESCRIPTION
**Pull request by FernandoS27**

### CODE REVIEW

File: src/video_core/engines/maxwell_3d.cpp [hunk 1]
```diff
@@ -500,6 +489,22 @@ Texture::FullTextureInfo Maxwell3D::GetStageTexture(Regs::ShaderStage stage,
     return tex_info;
 }
 
+Texture::FullTextureInfo Maxwell3D::GetStageTexture(Regs::ShaderStage stage,
+                                                    std::size_t offset) const {
+    auto& shader = state.shader_stages[static_cast<std::size_t>(stage)];
+    auto& tex_info_buffer = shader.const_buffers[regs.tex_cb_index];
```
Comment (lioncash) [inline src/video_core/engines/maxwell_3d.cpp]:
```suggestion
    const auto& tex_info_buffer = shader.const_buffers[regs.tex_cb_index];
```

File: src/video_core/const_buffer_accessor.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+#include <cstring>
+
+#include "common/common_types.h"
+#include "core/core.h"
+#include "video_core/engines/maxwell_3d.h"
+#include "video_core/gpu.h"
+#include "video_core/memory_manager.h"
+
+namespace Tegra {
```
Comment (lioncash) [inline src/video_core/const_buffer_accessor.h]:
```suggestion
namespace Tegra::ConstBufferAccessor {
```

File: src/video_core/shader/shader_ir.h [hunk 3]
```diff
@@ -195,9 +195,23 @@ enum class ExitMethod {
 
 class Sampler {
 public:
+    // Use this constructor for binded Samplers
```
Comment (ReinUsesLisp) [inline src/video_core/shader/shader_ir.h]: `bound` and `samplers`

File: src/video_core/shader/shader_ir.h [hunk 4]
```diff
@@ -219,6 +233,14 @@ class Sampler {
         return is_shadow;
     }
 
+    bool IsBindless() const {
+        return is_bindless;
+    }
+
+    std::pair<u32, u32> GetBindlessCBuf() const {
+        return {offset >> 32, offset & 0x00000000FFFFFFFFULL};
```
Comment (ReinUsesLisp) [inline src/video_core/shader/shader_ir.h]: This looks like a `static_cast<u32>(offset)`.

File: src/video_core/const_buffer_accessor.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace ConstBufferAccessor {
+
+template <typename T>
+T access(Tegra::Engines::Maxwell3D::Regs::ShaderStage stage, u64 const_buffer, u64 offset) {
```
Comment (lioncash) [inline src/video_core/const_buffer_accessor.h]: Does this really need to be in its own file? This seems like something that would make more sense to be part of the Maxwell3D engine's interface, given it's accessing its shader stages.

In any case, this shouldn't be using the `Core::System::GetInstance()` accessor directly. This should be taking the system object as a parameter.

**Comment by lioncash** [comment]
My bad if any comments overlap with Rodrigo's. My browser window didn't update when his review was submitted.

**Comment by FernandoS27** [comment]
Comments addressed.

**Pull request merged**